### PR TITLE
fix(http): do not attempt to autolink if previous character is often found in markup

### PR DESCRIPTION
**Pull request by trieloff**
### Initial checklist

*   [x] I read the support docs 
*   [x] I read the contributing guide 
*   [x] I agree to follow the code of conduct 
*   [x] I searched issues and couldn’t find anything (or linked relevant results below) 
*   [x] If applicable, I’ve added docs and tests

### Description of changes

- added a check to `previousHttp` to avoid `>` and `=`
- made sure that the previous check does not apply to `previousEmail`

see https://github.com/micromark/micromark-extension-gfm/pull/4

Porting the tests is a bit tricky, as they depend on the interplay with raw HTML parsing, so I'm only adding minimal tests here.

### CODE REVIEW

File: test/fixtures/previous.comment.md [hunk 1]
```diff
@@ -17,6 +17,10 @@ Can start after underscore *_https://a.b.
 
 Can start after tilde ~https://a.b.
 
+Cannot start after greater than >http://www.example.com
```
Comment (ChristianMurphy) [inline test/fixtures/previous.comment.md:20]: It appears to be linked on GitHub though :thinking: 

> Cannot start after greater than >http://www.example.com

File: test/fixtures/previous.comment.md [hunk 2]
```diff
@@ -17,6 +17,10 @@ Can start after underscore *_https://a.b.
...
+
+Cannot start after equals sign =http://www.example.com
```
Comment (ChristianMurphy) [inline test/fixtures/previous.comment.md:22]: same here

> Cannot start after equals sign =http://www.example.com

**Review by ChristianMurphy — COMMENTED** [review]
This appears to differ from GitHub's autolinking algorithm :thinking:

**Comment by trieloff** [comment]
The proper way would probably be to check if we are in an HTML `a` tag, but I'm not sure how to do that.

File: test/fixtures/previous.comment.html [hunk 3]
```diff
@@ -8,6 +8,8 @@ <h1>HTTP</h1>
 <p>Can start after asterisk *<a href="https://a.b">https://a.b</a>.</p>
 <p>Can start after underscore *_<a href="https://a.b">https://a.b</a>.</p>
 <p>Can start after tilde ~<a href="https://a.b">https://a.b</a>.</p>
+<p>Cannot start after greater than >http://www.example.com</p>
+<p>Cannot start after equals sign =http://www.example.com</p>
```
Comment (wooorm) [inline test/fixtures/previous.comment.html:12]: This file is generated from GitHub (`UPDATE=1 npm test`), I’m assuming you added these manually, and that the actual GH results are different?

**Comment by wooorm** [comment]
The goal of this package is to match GitHub, and GitHub does not do this. Instead, GitHub adds links inside links

**Pull request closed without merging**